### PR TITLE
[ADD] Order chatter task and activity

### PR DESCRIPTION
**Pull request by eilst**

### CODE REVIEW

File: project_event/models/project_task.py [hunk 1]
```diff
@@ -1012,3 +1012,46 @@ def get_confirmation_wizard(self, action):
     def check_task_state(task_state_in):
         return task_state_in in \
             ['draft', 'option', 'postponed', 'canceled']
+    
+    @api.multi
+    def _message_track(self, tracked_fields, initial):
+        mail_track = super()._message_track(tracked_fields, initial)
+        changes = mail_track[0]
+        tracking_value_ids = mail_track[1]
+        import ipdb; ipdb.set_trace()
```
Comment (eilst) [inline project_event/models/project_task.py]:
```suggestion
```

File: project_event/models/project_task.py [hunk 2]
```diff
@@ -1012,3 +1012,48 @@ def get_confirmation_wizard(self, action):
...
+        tracking_value_ids = mail_track[1]
+        if self.activity_task_type == 'activity':
+            tracking_value_ids = self.order_activity_fields(tracking_value_ids)
+        elif self.activity_task_type == 'task':
+            tracking_value_ids = self.order_task_fields(tracking_value_ids)
+        return changes, tracking_value_ids
+
+    @staticmethod
+    def order_activity_fields(tracking_values):
+        activity_fields_list = [
```
Comment (rimbendhaou) [inline project_event/models/project_task.py:1029]: The order of the fields is: 'task_state', 'name', 'code', 'responsible_id', 'partner_id', 'room_id', 'date_start', 'date_end', 'notes'
Only these fields should be tracked in the chatter, the others shouldnt appear

File: project_event/models/project_task.py [hunk 3]
```diff
@@ -1012,3 +1012,48 @@ def get_confirmation_wizard(self, action):
...
+                activity = tracking_values.__getitem__(k)
+                if activity.__getitem__(2).get('field')\
+                        == activity_fields_list[index]:
+                    activity_tracking_values.append(activity)
+        return activity_tracking_values
+
+    @staticmethod
+    def order_task_fields(tracking_values):
+        task_fields_list = [
+            'name', 'code', 'responsible_id', 'partner_id', 'project_id',
```
Comment (rimbendhaou) [inline project_event/models/project_task.py]: The order of the fields is: 'task_state', 'name', 'code', 'responsible_id', 'partner_id', 'date_start', 'date_end', 'notes', 'department_id', 'employee_ids'
Only these fields should be tracked in the chatter, the others shouldn't appear

**Review by rimbendhaou — CHANGES_REQUESTED** [review]

File: project_event/models/project_task.py (hunk 2 above)
Comment (eilst) [reply project_event/models/project_task.py:1029]: Done

File: project_event/models/project_task.py (hunk 3 above)
Comment (eilst) [reply project_event/models/project_task.py]: Done :+1:

**Review by rimbendhaou — APPROVED** [review]

File: project_event/models/project_task.py [hunk 4]
```diff
@@ -1012,3 +1012,58 @@ def get_confirmation_wizard(self, action):
...
+        activity_fields_list = [
+            'task_state',
+            'name',
+            'code',
+            'responsible_id',
+            'partner_id',
+            'room_id',
+            'date_start',
+            'date_end',
+            'notes'
```
Comment (eilst) [inline project_event/models/project_task.py]:
```suggestion
            'notes',
            'project_id',
            'category_id',
            'resource_type',
            'manager_id',
            'user_id',
            'client_type',
            'sector_id'
```

File: project_event/models/project_task.py [hunk 5]
```diff
@@ -1012,3 +1012,65 @@ def get_confirmation_wizard(self, action):
...
+            'task_state',
+            'name',
+            'code',
+            'responsible_id',
+            'partner_id',
+            'date_start',
+            'date_end',
+            'notes',
+            'department_id',
+            'employee_ids'
```
Comment (eilst) [inline project_event/models/project_task.py]:
```suggestion
            'employee_ids',
            'project_id',
            'category_id',
            'room_id',
            'resource_type',
            'user_id',
            'client_type',
            'sector_id',
            'rel_date_start'
```

File: project_event/models/project_task.py [hunk 6]
```diff
@@ -1012,3 +1012,73 @@ def get_confirmation_wizard(self, action):
...
+            'department_id',
+            'employee_ids',
+            'project_id',
+            'category_id',
+            'room_id',
+            'resource_type',
+            'user_id',
+            'client_type',
+            'sector_id',
+            'rel_date_start'
```
Comment (eilst) [inline project_event/models/project_task.py]:
```suggestion
            'rel_date_start',
            'rel_date_end',
            'report_done_required'
```

File: project_event/models/project_task.py [hunk 7]
```diff
@@ -1012,3 +1012,75 @@ def get_confirmation_wizard(self, action):
...
+            'resource_type',
+            'user_id',
+            'client_type',
+            'sector_id',
+            'rel_date_start',
+            'rel_date_end',
+            'report_done_required'
+        ]
+        task_tracking_values = []
+        for x in range(len(tracking_values)):
```
Comment (eilst) [inline project_event/models/project_task.py]:
```suggestion
        for x in range(len(task_fields_list)):
```

File: project_event/models/project_task.py [hunk 8]
```diff
@@ -1012,3 +1012,75 @@ def get_confirmation_wizard(self, action):
...
+            'project_id',
+            'category_id',
+            'resource_type',
+            'manager_id',
+            'user_id',
+            'client_type',
+            'sector_id'
+        ]
+        activity_tracking_values = []
+        for index in range(len(tracking_values)):
```
Comment (eilst) [inline project_event/models/project_task.py]:
```suggestion
        for index in range(len(activity_fields_list)):
```

**Pull request merged**